haskell/private/create_dummy_archive.bzl

```diff
@@ -0,0 +1,60 @@
+load("@bazel_skylib//:lib.bzl", "paths")
+load(":cc.bzl", "cc_interop_info")
+load(":private/context.bzl", "haskell_context")
+
+def _impl(ctx):
+  """Create empty archive so that GHC has some input files to work on during
+  linking.
+
+  See: https://github.com/facebook/buck/blob/126d576d5c07ce382e447533b57794ae1a358cc2/src/com/facebook/buck/haskell/HaskellDescriptionUtils.java#L295
+
+  Returns:
+    File, the created dummy archive.
+  """
+
+  # This is a bit of a hack to make `cc_interop_info` accept this rule’s `ctx`
+  ctx.attr.deps = []
```

Comment:
I thought I could just stub out the attribute without creating an extra field in `attrs`, but that is forbidden:

```
ERROR: /Users/distiller/project/haskell/private/create_dummy_archive.bzl:16:3: cannot assign to 'ctx.attr.deps'
```